extensions/amp-analytics/0.1/requests.js

```diff
@@ -258,15 +268,26 @@ export class RequestHandler {
    */
   constructExtraUrlParamStrs_(baseUrl, extraUrlParamStrsPromise) {
     return Promise.all(extraUrlParamStrsPromise).then(paramStrs => {
-      filterSplice(paramStrs, item => {return !!item;});
-      const extraUrlParamsStr = paramStrs.join('&');
+      const extraUrlParamStrs = {
+        requestUrl: baseUrl,
+        requestBody: '',
+      };
       let requestUrl;
-      if (baseUrl.indexOf('${extraUrlParams}') >= 0) {
-        requestUrl = baseUrl.replace('${extraUrlParams}', extraUrlParamsStr);
+      if (!this.useBody_) {
+        filterSplice(paramStrs, item => {return !!item;});
+        const extraUrlParams = paramStrs.join('&');
+        if (baseUrl.indexOf('${extraUrlParams}') >= 0) {
+          requestUrl = baseUrl.replace('${extraUrlParams}', extraUrlParams);
+        } else {
+          requestUrl =
+            appendEncodedParamStringToUrl(baseUrl, extraUrlParams);
+        }
+        extraUrlParamStrs.requestUrl = requestUrl;
       } else {
-        requestUrl = appendEncodedParamStringToUrl(baseUrl, extraUrlParamsStr);
+        extraUrlParamStrs.requestBody = paramStrs[0];
```

Comment:
Hmmm Does this means that `useBody` doesn't support batching, as we are only sending `paramStrs[0]` ?